src/harness/reference_models/ppa_model/ppa_ref_model.py

```diff
@@ -0,0 +1,82 @@
+#    Copyright 2016 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from concurrent.futures import ProcessPoolExecutor as Pool
+import multiprocessing
+from utils import *
+from compute import load_partials
+from reference_models.geo.vincenty import GeodesicPoint
+
+THRESHOLD = -96
+
+
+def __compute_points_gain_prop(device):
+  pool = Pool(multiprocessing.cpu_count())
```

Comment:
The pool is a common resource and should not be created for each call as it is expensive.
It can be created either as a global, or passed to the function (and created at the caller).

Also note that the caller is itself calling __compute_points_gain_prop() with a pool, sending every device calculation in parallel. It is a bad idea to nest such pool use, and can create issues, contention and no performance gain in fact.

Also note that the python multiprocessing pool is using system multi process as multithreading does not give any advantage in Python. Multi process means that instance of Python process are created multiple time, and there is some heavy  communication to pass the data from the master process to the child process. It is thus interesting for some heavy calculation, and not very small calculation.
I am not sure some of the calculation below are big enough to require such an heavy artillery.

If the performance is not so much an issue for these tests, then we could simply live with the top level multiprocessing at the device stage, and keep simple `map` here.